src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -680,55 +698,39 @@ export default {
 		// Checks whether the conversations list is scrolled all the way to the top
 		// or not
 		handleScroll() {
-			this.isScrolledToTop = this.$refs.container.scrollTop === 0
-		},
-		elementIsAboveViewpoint(container, element) {
-			return element.offsetTop < container.scrollTop
-		},
-		elementIsBelowViewpoint(container, element) {
-			return element.offsetTop + element.offsetHeight > container.scrollTop + container.clientHeight
+			this.isScrolledToTop = this.$refs.scroller.$el.scrollTop === 0
 		},
+
+		/**
+		 * Find position of unread conversation below viewport
+		 */
 		handleUnreadMention() {
-			this.unreadNum = 0
-			const unreadMentions = document.querySelectorAll('.unread-mention-conversation')
-			unreadMentions.forEach(x => {
-				if (this.elementIsBelowViewpoint(this.$refs.container, x)) {
-					if (this.unreadNum === 0) {
-						this.firstUnreadPos = x.offsetTop
+			this.$nextTick(() => {
+				this.unreadNum = 0
+				const lastConversationInViewport = this.$refs.scroller.getLastItemInViewportIndex()
+				for (let i = this.filteredConversationsList.length - 1; i > lastConversationInViewport; i--) {
+					if (this.filteredConversationsList[i].unreadMention) {
+						this.unreadNum = 1
+						this.firstUnreadIndex = i
+						return
 					}
-					this.unreadNum += 1
 				}
 			})
 		},
+
 		debounceHandleScroll: debounce(function() {
 			this.handleScroll()
 			this.handleUnreadMention()
 		}, 50),
 
-		scrollToConversation(token) {
-			this.$nextTick(() => {
-				// In Vue 2 ref on v-for is always an array and its order is not guaranteed to match the order of v-for source
-				// See https://github.com/vuejs/vue/issues/4952#issuecomment-280661367
-				// Fixed in Vue 3
-				// Temp solution - use unique ref name for each v-for element. The value is still array but with one element
-				// TODO: Vue3: remove [0] here or use object for template refs
-				const conversation = this.$refs[`conversation-${token}`]?.[0].$el
-				if (!conversation) {
-					return
-				}
+		async scrollToConversation(token) {
+			await this.$nextTick()
 
-				if (this.elementIsBelowViewpoint(this.$refs.container, conversation)) {
-					this.$refs.container.scrollTo({
-						top: conversation.offsetTop + conversation.offsetHeight * 2 - this.$refs.container.clientHeight,
-						behavior: 'smooth',
-					})
-				} else if (this.elementIsAboveViewpoint(this.$refs.container, conversation)) {
-					this.$refs.container.scrollTo({
-						top: conversation.offsetTop - conversation.offsetHeight,
-						behavior: 'smooth',
-					})
-				}
-			})
+			if (!this.$refs.scroller) {
+				return
+			}
+
+			await this.$refs.scroller.scrollToConversation(token)
```

Comment:
Hmm... I need to check it, thanks

Reply:
Well, the scrolling doesn't work on the `master` as well, because conversation can be joined before the list is loaded. But it helped me to find a regression (https://github.com/nextcloud/spreed/pull/10051#issuecomment-1683867869)